metagraph/src/common/vectors/bitmap.cpp

```diff
@@ -369,3 +369,52 @@ void bitmap_lazy::call_ones_in_range(uint64_t begin, uint64_t end,
             callback(i);
     }
 }
+
+
+////////////////////////////////////////////////////////////////
+//                         bitmap_generator                   //
+////////////////////////////////////////////////////////////////
+
+bitmap_generator::bitmap_generator(size_t size, bool value)
+      : bitmap_generator([value,size](const auto &index_callback) {
+                             if (value) {
+                                 for (size_t i = 0; i < size; ++i) {
+                                     index_callback(i);
+                                 }
+                             }
+                         }, size, size * value) {}
+
+bitmap_generator
+::bitmap_generator(std::function<void(const VoidCall<uint64_t>&)>&& generator,
+                   size_t size,
+                   size_t num_set_bits) noexcept
+      : size_(size), num_set_bits_(num_set_bits), generator_(std::move(generator)) {}
+
+bitmap_generator
+::bitmap_generator(bitmap&& base, std::function<uint64_t(uint64_t)>&& index_transformer,
+                   size_t size, size_t num_set_bits) noexcept
```

Comment:
Hard to read. Make each on a new line
```suggestion
::bitmap_generator(bitmap&& base,
                   std::function<uint64_t(uint64_t)>&& index_transformer,
                   size_t size,
                   size_t num_set_bits) noexcept
```